src/mock-doc/custom-element-registry.ts

```diff
@@ -81,14 +81,14 @@ export class MockCustomElementRegistry implements CustomElementRegistry {
     }
   }
 
-  whenDefined(tagName: string) {
+  whenDefined(tagName: string): Promise<CustomElementConstructor> {
     tagName = tagName.toLowerCase();
 
     if (this.__registry != null && this.__registry.has(tagName) === true) {
-      return Promise.resolve();
+      return this.__registry.get(tagName).cstr;
```

Comment:
I'm not completely sure what the type of `cstr` is, but assuming it's `CustomElementConstructor` I would expect this to still return a promise:

```suggestion
      return Promise.resolve<CustomElementConstructor>(this.__registry.get(tagName).cstr);
```

Reply:
Good call - technically it may not be a promise. Updated in https://github.com/ionic-team/stencil/pull/3205/commits/b365de1984f4e948f5acee6f5da0ba1e61508fd2